src/chainparams.h

```diff
@@ -145,6 +145,8 @@ class CChainParams
     int HeightLightZerocoin() const { return nHeightLightZerocoin; }
     int HeightEnforceBlacklist() const { return nHeightEnforceBlacklist; }
 
+    uint32_t PowUpdateTimestamp() const { return nPowUpdateTimestamp; }
```

Comment:
all time is using uint32_t. Up to you though.

Reply:
I agree with the uint32_t.  "int" is overused and has been known to trigger problems when using blindly for a unsigned value.  If it's unsigned, it should be unsigned.  If it's 32 bit, it should be specified as such.  I would argue (weakly) for a refactor to change ints (especially for things like height) to "uint32_t".  However, since the "problem" doesn't come in until you're after 2.1b blocks (or about 4000 years at a minute per block) it's not something i'm going to push hard for ;)

However for an epoch time, the "problem" comes much sooner; January 2038.  So uint32_t is much more important (and yes, since 1999 I have been way more concerned about when Epoch time exceeds 31 bits then "y2k" bugs)

Reply:
Sure, I'm ok with this.